qa/vagrant/src/main/java/org/elasticsearch/packaging/util/Platforms.java

```diff
@@ -28,41 +28,53 @@ public static boolean isDPKG() {
         if (WINDOWS) {
             return false;
         }
-        return new Shell().bashIgnoreExitCode("which dpkg").isSuccess();
+        return new Shell().runIgnoreExitCode("which dpkg").isSuccess();
     }
 
     public static boolean isAptGet() {
         if (WINDOWS) {
             return false;
         }
-        return new Shell().bashIgnoreExitCode("which apt-get").isSuccess();
+        return new Shell().runIgnoreExitCode("which apt-get").isSuccess();
     }
 
     public static boolean isRPM() {
         if (WINDOWS) {
             return false;
         }
-        return new Shell().bashIgnoreExitCode("which rpm").isSuccess();
+        return new Shell().runIgnoreExitCode("which rpm").isSuccess();
     }
 
     public static boolean isYUM() {
         if (WINDOWS) {
             return false;
         }
-        return new Shell().bashIgnoreExitCode("which yum").isSuccess();
+        return new Shell().runIgnoreExitCode("which yum").isSuccess();
     }
 
     public static boolean isSystemd() {
         if (WINDOWS) {
             return false;
         }
-        return new Shell().bashIgnoreExitCode("which systemctl").isSuccess();
+        return new Shell().runIgnoreExitCode("which systemctl").isSuccess();
     }
 
     public static boolean isSysVInit() {
         if (WINDOWS) {
             return false;
         }
-        return new Shell().bashIgnoreExitCode("which service").isSuccess();
+        return new Shell().runIgnoreExitCode("which service").isSuccess();
+    }
+
+    public static void onWindows(Runnable action) {
+        if (WINDOWS) {
+            action.run();
+        }
+    }
+
+    public static void onLinux(Runnable action) {
```

Comment:
I topically declare a `@FunctionalInterface` for this just to avoid confusion. 
Just by looking at this one could assume that the action is carried out in a new thread.

Reply:
>one could assume that the action is carried out in a new thread

That's a good point, I think that's worth adding another type for